source/DasBlog.Web.Repositories/SiteSecurityRepository.cs

```diff
@@ -9,151 +9,151 @@
-        {
-            throw new NotImplementedException();
-        }
-
-        public string Encrypt(string cleanString)
-        {
-            if (!IsCleanStringEncrypted(cleanString))
-            {
-                Byte[] clearBytes = Encoding.Unicode.GetBytes(cleanString);
-
-                MD5 md5 = MD5CryptoServiceProvider.Create();
-                Byte[] hashedBytes = md5.ComputeHash(clearBytes);
-
-                return BitConverter.ToString(hashedBytes);
-            }
-            else
-                return cleanString;
-        }
-
-        public UserToken GetToken(string userName)
-        {
-            User user = GetUser(userName);
-            return user?.ToToken();
-        }
-
-        public bool IsCleanStringEncrypted(string cleanString)
-        {
-            return (cleanString.Length == 47 && cleanString.Replace("-", String.Empty).Length == 32);
-        }
-
-        public void Login(UserToken token, string userName)
-        {
-            if (token == null)
-            {
-                // SiteSecurity.LogFailure(userName);
-            }
-            else
-            {
-                // SiteSecurity.LogSuccess(token.Name);
-                GenericIdentity identity = new GenericIdentity(token.Name, "Custom");
-                GenericPrincipal principal = new GenericPrincipal(identity, new string[] { token.Role });
-                // HttpContext.Current.User = principal;
-                System.Threading.Thread.CurrentPrincipal = principal;
-            }
-        }
-
-        public UserToken Login(string userName, string password)
-        {
-            UserToken token = null;
-
-            User user = GetUser(userName);
-
-            if (user != null && user.Active)
-            {
-                //Make sure password is encrypted
-                if (!IsCleanStringEncrypted(password))
-                {
-                    password = Encrypt(password);
-                }
-
-                if ((IsCleanStringEncrypted(user.Password) && user.Password == password) || (Encrypt(user.Password) == password))
-                {
-                    token = user.ToToken();
-                }
-            }
-
-            Login(token, userName);
-
-            return token;
-        }
-
-        public UserToken Login(string userName, string clientHash, string challenge)
-        {
-            throw new NotImplementedException();
-        }
-
-        public void SetPassword(string userName, string password)
-        {
-            throw new NotImplementedException();
-        }
-    }
+	public class SiteSecurityRepository : ISiteSecurityRepository
+	{
+		private IPrincipal _principal;
+		private IDasBlogSettings _dasBlogSettings;
+
+		public SiteSecurityRepository(IPrincipal principal, IDasBlogSettings dasBlogSettings)
+		{
+			_dasBlogSettings = dasBlogSettings;
+			_principal = principal;
+		}
+
+		public bool IsValidContributor { get { return _principal.IsInRole(null); } }
```

Comment:
This should be looks something like this:

public bool IsValidContributor { get { return (_principal.IsInRole(Role.Administrator) || _principal.IsInRole(Role.Contributor)); } }